__tests__/app.test.js

```diff
@@ -131,24 +127,60 @@ describe("task 6 - /api/articles/:article_id/comments",()=>{
             })
         
         })
-        test("Error:404 when article_id is valid but not found",()=>{
+        test("error:404 when article_id is valid but not found",()=>{
             return request(app).get("/api/articles/999/comments")
             .expect(404).then(({body})=>{   
                 expect(body.msg).toBe('article does not exist')
                 })
         })
-        test("Error:400 when article_id is invalid",()=>{
+        test("error:400 when article_id is invalid",()=>{
             return request(app).get("/api/articles/not-a-value/comments")
             .expect(400).then(({body})=>{
                 expect(body.msg).toBe('Bad request')
                 })
         })
-        test("Status:200 when article_id is valid but no comments",()=>{
+        test("status code:200 when article_id is valid but no comments",()=>{
             return request(app).get("/api/articles/13/comments")
             .expect(200).then(({body})=>{
                 expect(body.msg).toBe('No comments for article 13')
                 })
         })
 
     })
+})
+describe("task 7 - /api/articles/:article_id/comments",()=>{
+    describe("POST",()=>{
+        test("status code:201 and returns a completed comment entry",()=>{
+            return request(app).post("/api/articles/2/comments").send({userName: "icellusedkars",body:"text"})
+            .expect(201).then(({body})=>{
+                expect(body.newPost).toMatchObject({
+                    comment_id:19,
+                    author: "icellusedkars",
+                    body:"text",
+                    article_id: 2,
+                    votes:0,
+                    created_at: expect.stringMatching(datePattern)
+                })
+            })
+            
+        })
+        test("error:400 when post request is bad - missing value",()=>{
+            return request(app).post("/api/articles/2/comments").send({userName: "icellusedkars"})
+            .expect(400).then(({body})=>{
+                expect(body.msg).toBe('Bad request')
+                })
+        })
+        test("error:400 when post request is bad - wrong datatype",()=>{
+            return request(app).post("/api/articles/2/comments").send({userName: "icellusedkars",body:1234})
+            .expect(400).then(({body})=>{
+                expect(body.msg).toBe('Bad request')
+                })
+        })
+        test("error:403 when post request is forbidden (username is not present in users table - violates constraints)",()=>{
+            return request(app).post("/api/articles/2/comments").send({userName: "iNeedSleep", body: "text"})
+            .expect(403).then(({body})=>{
+                expect(body.msg).toBe('Forbidden - possible table constraint violation, check constraints of target table')
+            })
+        })
```

Comment:
I'd recommend removing the tests for the 403's and instead adding the following:

- Status 404: Not found, username does not exist
- Status 404: Not found, article_id does not exist